From b572ea0258fce64838aa337d92aec3b78d47a6f3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:23 -0400
Subject: [powerpc] Fix hw breakpoints on !HAVE_HW_BREAKPOINT configurations

Message-id: <1410545655-205645-275-git-send-email-dzickus@redhat.com>
Patchwork-id: 94286
O-Subject: [RHEL7 PATCH 274/626] powerpc: Fix hw breakpoints on !HAVE_HW_BREAKPOINT configurations
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 1c430c06d0ce871f36b2af504d45a07356e44800
Author: Andreas Schwab <schwab@linux-m68k.org>
Date:   Tue Jan 21 23:24:02 2014 +0100

    powerpc: Fix hw breakpoints on !HAVE_HW_BREAKPOINT configurations

    This fixes a logic error that caused a failure to update the hw breakpoint
    registers when not using the hw-breakpoint interface.

    Signed-off-by: Andreas Schwab <schwab@linux-m68k.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/process.c b/arch/powerpc/kernel/process.c
index db3be92..9d1ecbb 100644
--- a/arch/powerpc/kernel/process.c
+++ b/arch/powerpc/kernel/process.c
@@ -833,7 +833,7 @@ struct task_struct *__switch_to(struct task_struct *prev,
  * schedule DABR
  */
 #ifndef CONFIG_HAVE_HW_BREAKPOINT
- if (unlikely(hw_brk_match(&__get_cpu_var(current_brk), &new->thread.hw_brk)))
+ if (unlikely(!hw_brk_match(&__get_cpu_var(current_brk), &new->thread.hw_brk)))
   set_breakpoint(&new->thread.hw_brk);
 #endif /* CONFIG_HAVE_HW_BREAKPOINT */
 #endif
-- 
1.7.1